datastore/migrations/core/migrater_memory.py

```diff
@@ -0,0 +1,141 @@
+                + f"Given migration index of import data: {self.start_migration_index} "
+                + f"Current backend migration index: {self.target_migration_index}"
+            )
+        else:
+            self.run_actual_migrations()
+
+        return False
+
+    def run_actual_migrations(self) -> None:
+        position = RawPosition(
+            position=1,
+            migration_index=self.start_migration_index,
+            timestamp=datetime.now(),
+            user_id=0,
+            information=None,
+        )
+        last_position_value = 0
+        self.migrate_position(position, last_position_value)
+
+    def migrate_position(
+        self, position: RawPosition, last_position_value: Position
+    ) -> None:
+        """
+        Used for in memory migration of import data.
+        All import data will be imported at one position.
+        """
+        migration_index = position.migration_index
+
+        self.logger.info(
+            f"Migrate import data from MI {migration_index} to MI {self.target_migration_index} ..."
+        )
+        for source_migration_index in range(
+            migration_index, self.target_migration_index
+        ):
+            target_migration_index = source_migration_index + 1
+            self.logger.debug(
+                f"\tRunning migration with target migration index {target_migration_index}"
+            )
+            old_accessor, new_accessor = self.get_accessors(
+                last_position_value,
+                source_migration_index,
+                target_migration_index,
+                position.position,
+                False,
+            )
+
+            migration = self.migrations[target_migration_index]
+
+            old_events = self.import_create_events
+            new_events = migration.migrate(
+                old_events, old_accessor, new_accessor, position.to_position_data()
+            )
+            self.import_create_events = new_events
+
+    def get_accessors(
+        self,
+        last_position_value: Position,
+        source_migration_index: int,
+        target_migration_index: int,
+        position: Position,
+        is_last_migration_index: bool,
+    ) -> Tuple[MigrationKeyframeModifier, MigrationKeyframeModifier]:
+        old_accessor = InitialMigrationKeyframeModifier(
+            self.connection,
+            last_position_value,
+            source_migration_index,
+            position,
+        )
+        old_accessor.models.update(self.imported_models)
+        old_accessor.deleted.update({key: False for key in self.imported_models})
+        new_accessor = InitialMigrationKeyframeModifier(
+            self.connection,
+            last_position_value,
+            target_migration_index,
+            position,
+        )
+        new_accessor.models.update(self.imported_models)
+        new_accessor.deleted.update({key: False for key in self.imported_models})
```

Comment:
Maybe outsource this into its own function to deduplicate the code?

Reply:
Done